### PR TITLE
fix info message when max training time reached

### LINKED ISSUE #7778: Elapsed time not shown correctly when max_time is reached
## 🐛 Bug

When setting the `max_time` flag in the Trainer and the time is reached, the elapsed time is not displayed correctly in the trainer stopping message. Happens both when running the script through the terminal or in a jupyter notebook.

### To Reproduce
Reproduced with the BoringModel:
```python3
def test_x():
    # init model
    model = BoringModel()

    # Initialize a trainer
    trainer = pl.Trainer(
        max_epochs=1000,
        max_time="00:00:00:15"
    )

    # Train the model ⚡
    trainer.fit(model, train, val)
```

Output:
`Time limit reached. Elapsed time is <bound method Timer.time_elapsed of <pytorch_lightning.callbacks.timer.Timer object at 0x2b3101e19320>>. Signaling Trainer to stop.
`
### Expected behavior
Elapsed time should be displayed.

### Environment

```
* CUDA:
        - GPU:
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
        - available:         True
        - version:           10.2
* Packages:
        - numpy:             1.19.0
        - pyTorch_debug:     False
        - pyTorch_version:   1.8.1+cu102
        - pytorch-lightning: 1.3.3
        - tqdm:              4.61.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                -
        - processor:         x86_64
        - python:            3.6.9
        - version:           #1 SMP Wed May 13 10:45:47 CDT 2020
```

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #7778

Fixes incorrect formatting of time remaining.
Now shows a message properly formatted:

> Time limit reached. Elapsed time is 0:00:03. Signaling Trainer to stop.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: tests/callbacks/test_timer.py [hunk 1]
```diff
@@ -106,9 +106,12 @@ def test_timer_stops_training(tmpdir):
         max_epochs=1000,
         callbacks=[timer],
     )
-    trainer.fit(model)
+    with caplog.at_level(logging.INFO):
+        trainer.fit(model)
     assert trainer.global_step > 1
     assert trainer.current_epoch < 999
+    assert "Time limit reached." in caplog.text
+    assert "Signaling Trainer to stop." in caplog.text
```
Comment (Borda) [inline tests/callbacks/test_timer.py:114]:
```suggestion
    assert re.match(r"Time limit reached. Elapsed time is .*\. Signaling Trainer to stop.", caplog.text)
```

**Review by carmocca — APPROVED** [review]

File: tests/callbacks/test_timer.py (hunk 1 above)
Comment (Borda) [reply tests/callbacks/test_timer.py:114]: @justusschock mind rather use reg expresions?

Comment (justusschock) [reply tests/callbacks/test_timer.py:114]: @awaelchli ^^

Comment (awaelchli) [reply tests/callbacks/test_timer.py:114]: @Borda it wouldn't work.     

Not even 

`assert re.match(r".*Time limit reached.*", caplog.text)`

wants to match.

Comment (Borda) [reply tests/callbacks/test_timer.py:114]:
```python
import re
m = "Time limit reached. Elapsed time is any text here I want. Signaling Trainer to stop."
assert re.match(r"Time limit reached. Elapsed time is .*\. Signaling Trainer to stop.", m)
```
or what is in the `caplog.text`?
